dcos_test_utils/jobs.py

```diff
@@ -97,9 +97,10 @@ def _wait_for_run_completion(j_id: str, r_id: str) -> bool:
                     log.info('Job run {} finished.'.format(r_id))
                     return True
                 else:
-                    raise requests.HTTPError(
+                    log.warning(
                         'Waiting for job run {} to be finished, but history for that job run is not available'
-                        .format(r_id), response=rc)
```

Comment:
Nitpick: My only thought here is if there might be some useful information to be gleaned via `response` (reason, text, elapsed, etc), but that is more of an enhancement than vital to this patch.

Reply:
@jbarrick-mesosphere  - this is changing the behavior! Is history available only doing a look up.

Why would a retry of the field that is already obtained help?

Reply:
@jbarrick-mesosphere  - I digged in further. I think the retry will be around `_is_history_available` which is doing a details called (which is easy to miss as http call). We could try the try on the above method, and that's my understanding of this comment too - https://github.com/dcos/dcos-test-utils/pull/62#pullrequestreview-166533123 

If you retry here, the run details is going to be tried again in addition to try only for history endpoint.

Reply:
@orsenthil this fixes the bug that is described in DCOS_OSS-4877. It is not a change in behavior (beyond correcting us to the behavior that the original author expected). It's not an issue if we retry both calls, but I can update it to clean it up a little bit.

Reply:
It was easy to miss the goal if we do a retry on two calls. We seem to have already gotten the `_details` per this method, but seems we were doing an extra call too. Please see if updating other method is easier or makes it easier to reason. If not, this is fine.

Reply:
I'm having trouble getting it to work there -I think it is making it more complex since we have nested retries. IMO we should just go with it as-is.